projects/ghostscript/build.sh

```diff
@@ -56,9 +56,47 @@ $CXX $CXXFLAGS $CUPS_LDFLAGS -std=c++11 -I. \
     $CUPS_LIBS \
     $LIB_FUZZING_ENGINE bin/gs.a
 
+$CXX $CXXFLAGS $CUPS_LDFLAGS -std=c++11 -I. \
+    -DMULTIPLE_COLORS \
+    $SRC/gstoraster_fuzzer.cc \
+    -o "$OUT/gstoraster_fuzzer_all_colors" \
+    -Wl,-rpath='$ORIGIN' \
+    $CUPS_LIBS \
+    $LIB_FUZZING_ENGINE bin/gs.a
+
+$CXX $CXXFLAGS $CUPS_LDFLAGS -std=c++11 -I. \
+    -DPDF_TARGET \
+    $SRC/gstoraster_fuzzer.cc \
+    -o "$OUT/pdf_fuzzer" \
+    -Wl,-rpath='$ORIGIN' \
+    $CUPS_LIBS \
+    $LIB_FUZZING_ENGINE bin/gs.a
+
+# Create PDF seed corpus
+zip -j "$OUT/pdf_fuzzer_seed_corpus.zip" $SRC/pdf_seeds/*
+
+# Create corpus for all_color_fuzzer. Only use seeds of a few KB in size.
+mkdir -p "$WORK/all_color_seeds"
+for f in examples/ridt91.eps examples/snowflak.ps $SRC/pdf_seeds/pdf.pdf; do
+  printf "\x01" | cat - "$f" > tmp_file.txt
+  mv tmp_file.txt $f
+  s=$(sha1sum "$f" | awk '{print $1}')
+  cp "$f" "$WORK/all_color_seeds/$s"
+done
+zip -j "$OUT/gstoraster_fuzzer_all_colors_seed_corpus.zip" "$WORK"/all_color_seeds/*
```

Comment:
Does it matter that we keep these separate?  Isn't oss-fuzz supposed to figure out what's useful and what's not?

Reply:
The reason I separate is because the existing fuzzer has some quite large seeds (333KB) which is huge for fuzzing. This can be difficult for a fuzzer to handle in that it will have a large space to randomise where many mutations may not have any effect. Note that one of the main issues with the existing fuzzer is that it's very slow, i.e. 2 exec/sec. As such, we should aim to maximise the value of each mutation tried.

I would like to experiment with seeds that are of smaller size without changing the settings of the existing fuzzer, and this is the reason I separated them. The goal is to both improve fuzzing's ability to mutate relevant bits and also decrease execution time of ghostscript.

Reply:
The reason that the seed is so large is because it's a whole PDF file.  One of the difficulties with the fuzzer is that it currently only processes whole PDFs because of the API that we're using.  We'd need to write different tests if you want to do more granular fuzzing (which uses some internal code).

Reply:
I think the PDF file itself is also relatively large (at least relative to how small PDF files can get). I included a new seed which is 35kb while the existing one is 350kb -- let's see what the effects it has.